builder/pattern_assembler.js

```diff
@@ -63,7 +63,7 @@
         if (config.debug) {
           console.log('rendering full oPattern: ' + pattern.name);
         }
-        return pattern.render(data, partials);
+        return pattern.render(pattern, data, partials);
```

Comment:
Question: why do we need to pass the pattern in this way? Since .render() is a method of the pattern itself, the pattern object should be accessible in .render() as `this`.